src/nbl/video/CVKLogicalDevice.h

```diff
@@ -32,27 +54,115 @@ class CVKLogicalDevice final : public ILogicalDevice
+        vk_createInfo.imageExtent = { params.width, params.height };
+        vk_createInfo.imageArrayLayers = params.arrayLayers;
+        vk_createInfo.imageUsage = static_cast<VkImageUsageFlags>(params.imageUsage);
+        vk_createInfo.imageSharingMode = static_cast<VkSharingMode>(params.imageSharingMode);
+        vk_createInfo.queueFamilyIndexCount = static_cast<uint32_t>(params.queueFamilyIndices->size());
+        vk_createInfo.pQueueFamilyIndices = params.queueFamilyIndices->data();
+        vk_createInfo.preTransform = VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR; // Todo(achal)     
+        vk_createInfo.compositeAlpha = VK_COMPOSITE_ALPHA_OPAQUE_BIT_KHR; // Todo(achal)
+        vk_createInfo.presentMode = static_cast<VkPresentModeKHR>(params.presentMode);
+        vk_createInfo.clipped = VK_TRUE; // Todo(achal)
+        vk_createInfo.oldSwapchain = VK_NULL_HANDLE; // Todo(achal)
+
+        VkSwapchainKHR vk_swapchain;
+        if (vkCreateSwapchainKHR(m_vkdev, &vk_createInfo, nullptr, &vk_swapchain) != VK_SUCCESS)
+            return nullptr;
+
+        uint32_t imageCount;
+        VkResult retval = vkGetSwapchainImagesKHR(m_vkdev, vk_swapchain, &imageCount, nullptr);
+        if ((retval != VK_SUCCESS) && (retval != VK_INCOMPLETE)) // Todo(achal): Would there be a need to handle VK_INCOMPLETE separately?
+            return nullptr;
+
+        assert(imageCount <= MAX_SWAPCHAIN_IMAGE_COUNT);
+
+        VkImage vk_images[MAX_SWAPCHAIN_IMAGE_COUNT];
+        retval = vkGetSwapchainImagesKHR(m_vkdev, vk_swapchain, &imageCount, vk_images);
+        if ((retval != VK_SUCCESS) && (retval != VK_INCOMPLETE)) // Todo(achal): Would there be a need to handle VK_INCOMPLETE separately?
+            return nullptr;
+
+        ISwapchain::images_array_t images = core::make_refctd_dynamic_array<ISwapchain::images_array_t>(imageCount);
+
+        uint32_t i = 0u;
+        for (auto& image : (*images))
+        {
+            CVulkanForeignImage::SCreationParams creationParams;
+            creationParams.arrayLayers = params.arrayLayers;
+            creationParams.extent = { params.width, params.height, 1u };
+            creationParams.flags = static_cast<CVulkanForeignImage::E_CREATE_FLAGS>(0); // Todo(achal)
+            creationParams.format = params.surfaceFormat.format;
+            creationParams.mipLevels = 1u;
+            creationParams.samples = CVulkanImage::ESCF_1_BIT; // Todo(achal)
+            creationParams.type = CVulkanImage::ET_2D;
+
+            image = core::make_smart_refctd_ptr<CVulkanForeignImage>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), std::move(creationParams),
+                vk_images[i++]);
+        }
+
+        return core::make_smart_refctd_ptr<CVKSwapchain>(
+            core::smart_refctd_ptr<CVKLogicalDevice>(this), std::move(params),
+            std::move(images), vk_swapchain);
     }
```

Comment:
swapchains should create themselves (enables spoofs/usermade swapchains), this method has to go from `ILogicalDevice`

Reply:
I guess this rolls into #246